docs/ADRs/016-use-launch-darkly-for-feature-flags.md

```diff
@@ -0,0 +1,60 @@
+
+Our application, much like others built off the [CMS MacPro QuickStart](https://github.com/CMSgov/macpro-quickstart-serverless), uses a continuous deployment approach. This means every merged code change that passes through all stages of the production pipeline (including extensive automated tests) is released to users. This approach is an excellent way to accelerate both development and the feedback loop with users. However, as new large features are built out in the application, we would like a way to prevent users from attempting to use partially implemented functionality as it rolls out over multiple sprint cycles.
+
+In other words: we want to be able to **selectively show features to users, while still getting the benefits of continuous deployment**. We are also interested in the ability to show different versions of the app to different types of users, e.g. be able to test a new feature only with users from the state of California or only with users that are part of a predetermined test group. A common way to approach this technical problem is through the use of [feature flags](https://martinfowler.com/articles/feature-toggles.html). There are several approaches to consider.
+
+## Considered Alternatives
+
+* Do nothing
+* Manage ourselves by using environment variables to toggle features on and off
+* Use AWS feature flag management tools
+* Use Launch Darkly
+
+## Decision Outcome
+
+Use LaunchDarkly because of the robust feature flag management tools that will allow our multidisciplinary team to effectively test, validate, and selectively expose new, complex features.
+
+## Pros and Cons of the Alternatives
+
+### Do nothing
+
+* `+` No change, no extra dependencies. Continue to release code to users as it is ready and users will see changes reflected in their environments.
+* `+` No communication needed around what code is available on what environment. If its merged,  it is immediately useable in the higher environments.
+* `-` Incremental changes due to continuous delivery, including incomplete work, is visible to users. There are already cases where this has become confusing or resulted in a broken user experience.
+* `-` A high level of coordination is needed between the team to make sure higher environments are is a good state before testing new users with features.
+* `-` No ability to selectively test groups of changes together on production without also making them immediately visible to users.
+
+### Manage ourselves by using environment variables to toggle features on and off
+
+* `+` We can hide groups of changes behind an environment flag and make sure it is tested together before making it visible to users.
+* `+` We already use environment variables to configure many parts of the application.
+* `+` JS libraries exist to make this easier such as [Unleash](https://github.com/Unleash)
+* `-` Does not de-couple the act of delivering code from the act of enabling a new feature. A release is still required. 
```

Comment:
To clarify, so this means to "turn on" a feature that is hidden due to environment variables would require an additional code change and release, correct?

Reply:
Correct.